fs/osfs.py

```diff
@@ -588,7 +588,7 @@ def geturl(self, path, purpose="download"):
         # type: (Text, Text) -> Text
         if purpose != "download":
             raise NoURL(path, purpose)
-        return "file://" + self.getsyspath(path)
+        return "file://" + self.getsyspath(path).replace('\\', '/')
```

Comment:
I may be wrong, but I'm not sure replacing double-backwards slashes with a single slash is sufficient to have a proper URL, the drive prefix should be URL-encoded as well.

Reply:
without the fix, geturl() gives: `file://D:\a\1\s\tests\fixtures\template`. with the fix, it gives `file://D:/a/1/s/tests/fixtures/template`, which fs.open_fs() would happily 'eat'.

Reply:
@chfw I believe @althonos is correct and more work is needed here - see e.g. `https://blogs.msdn.microsoft.com/ie/2006/12/06/file-uris-in-windows/` for how to construct "proper" file URIs on Windows

Reply:
Current change won't support dying IE because `file:///C:/exampleㄓ.txt` -> `file:///C:/example%E3%84%93.txt` by python quote function.

But Microsoft insisted the following for IE browser:
```
Incorrect: file:///C:/example%E3%84%93.txt
Correct: file:///C:/exampleㄓ.txt
```

Reply:
Here is the reason why IE insists on unicode over percentage quoted strings: https://blogs.msdn.microsoft.com/ieinternals/2014/04/22/unicode-in-url-changes-for-ie11/

But the rest of the world has followed: Berners-Lee's RFC 3986: https://tools.ietf.org/html/rfc3986#page-12

Reply:
Here is death ultimatum from Microsoft for IE: https://www.techradar.com/news/microsoft-internet-explorer-dead-1995-to-2019-rip

Definition by 2020, IE die out with python 2.

Reply:
I'm happy to not support IE. Modern browsers should be our target.

Reply:
Chrome iterated till version 76 now since 2008.